eras/allegra/impl/cardano-ledger-allegra.cabal

```
@@ -52,7 +52,7 @@ library
         cardano-crypto-class,
         cardano-ledger-binary,
         cardano-ledger-core >=1.0,
-        cardano-ledger-shelley >=1.0,
+        cardano-ledger-shelley:{cardano-ledger-shelley, testlib} >=1.0,
```

That is the point of teslibs, so production code does not depend on code related to testing
```suggestion
        cardano-ledger-shelley >=1.0,
```